libmatrix.c

```diff
@@ -54,6 +55,33 @@ static const char *matrixprpl_list_icon(PurpleAccount *acct, PurpleBuddy *buddy)
 }
 
 /**
+ * Callback for when joining a room fails, create the room
+ */
+static void room_create_callback(MatrixConnectionData *conn,
```

Comment:
Please could you find a better name for this - `get_roomid_by_alias_failed_callback` or something.